libs/components/packages/src/schematics/migrations/update-9/move-page-component/move-page-component.schematic.ts

```diff
@@ -0,0 +1,143 @@
+              if (pageImport.parent.elements.length > 1) {
+                // Yes. Remove only one named import.
+                const index = pageImport.parent.elements.findIndex(
+                  (element) => element.name.getText() === pageImport.getText()
+                );
+
+                // Find the start and end of the import.
+                let importStart: number;
+                let importEnd: number;
+                if (index > 0) {
+                  // Not the first import.
+                  // Remove text from the end of the previous import to the end of the import.
+                  importStart = (
+                    pageImport.parent.elements.at(
+                      index - 1
+                    ) as ts.ImportSpecifier
+                  ).getEnd();
+                  importEnd = (
+                    pageImport.parent.elements.at(index) as ts.ImportSpecifier
+                  ).getEnd();
+                } else {
+                  // First import.
+                  // Remove text from the start of the import to the start of the next import.
+                  importStart = (
+                    pageImport.parent.elements.at(index) as ts.ImportSpecifier
+                  ).getStart();
+                  importEnd = (
+                    pageImport.parent.elements.at(
+                      index + 1
+                    ) as ts.ImportSpecifier
+                  ).getStart();
+                }
+
+                // Remove the import.
+                removeRecorder.remove(importStart, importEnd - importStart);
+              } else {
+                // No. Remove the whole import declaration.
+                const importDeclaration = pageImport.parent.parent.parent;
+                removeRecorder.remove(
+                  importDeclaration.getFullStart(),
+                  importDeclaration.getFullWidth()
+                );
+              }
+
+              // Commit the update.
+              tree.commitUpdate(removeRecorder);
+            });
+          }
+        } else if (path.endsWith('.component.html')) {
+          // Update <sky-page layout="auto"> to remove the layout attribute.
+          const content = tree.readText(path);
+          // Based on code search, any use of `layout="auto"` matches this pattern.
+          const search = '<sky-page layout="auto"';
+          const skyPageIndex = content.indexOf(search);
+          if (skyPageIndex > -1) {
+            const recorder = tree.beginUpdate(path);
+            // Remove ' layout="auto"'.
+            recorder.remove(skyPageIndex + 9, 14);
+            tree.commitUpdate(recorder);
+          }
+        }
+      });
+    });
+
+    // Add the dependency to @skyux/pages.
+    if (addDependency) {
+      const pages = getPackageJsonDependency(tree, '@skyux/pages');
+      if (!pages) {
+        const layout = getPackageJsonDependency(tree, '@skyux/layout');
+        if (layout) {
+          addPackageJsonDependency(tree, {
```

Comment:
ok this is the only place I'm getting a little lost. `addDependency` was set to true if there were any imports from the pages library. Makes sense. They might already have that dependency, so first we get it and then only perform adding logic if they don't have it. Makes sense.

But then we get the layout package, and only add pages if they have layout? I suppose they would always have it since they were using that previously, and that gives the version number. Should we remove the layout dependency if they don't have any other imports from that library? Admittedly that sounds like it would be a bit tricky/overkill.

Reply:
If they don't have layout it's not clear how we got here. It's using the version number from layout. It could have used core. Since pages depends on layout it's ok to leave it.